### PR TITLE
Add an "utility" sub to check if an ip address is private or not

### PR DESCRIPTION
**Pull request by bigio**
Add a sub to check if an ip address is private or not, to be used to check if a $RelayAddr is local to our network or not.

### CODE REVIEW

File: mimedefang.pl.in [hunk 1]
```diff
@@ -7771,6 +7771,40 @@ sub read_config($) {
...
+#***********************************************************************
+sub is_ip_private($) {
+  my ($ip) = @_;
+
+  $private_ip = qr{^(?:
+    127\.|
+    0?10\.|
+    172\.0?1[6-9]\.|
+    172\.0?2[0-9]\.|
+    172\.0?3[0-2]\.|
```
Comment (dfskoll) [inline mimedefang.pl.in:7797]: Should be: 172\.0?3[01]\.

Comment (dfskoll) [reply mimedefang.pl.in:7797]: Also, we shouldn't be reinventing the wheel.  We could just use Net::IP, which has an ip_iptype function that returns "Public", "Private" or "Reserved", and also handles things like IPV4-encapsulated-in-IPV6 addresses.

**Comment by bigio** [comment]
I decided not to use Net::IP because I did not want to add a dependency just for a simple sub.
If Net::IP would be used for other code as well, it makes sense to use Net::IP->iptype().

**Comment by dfskoll** [comment]
Sure, but as you see, even in a simple sub, it's easy to introduce a bug (the off-by-one error for the 172.16/12 network.)  In almost all cases, IMO, it's better to reuse tested modules than write our own.

**Comment by bigio** [comment]
adding a dependency just to add a line of code is not worth imho nor reinventing the wheel, it would be better to maintain similar code on my mimedefang-filter code and drop this PR.

**Pull request closed without merging**